src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -187,39 +190,53 @@ export class IntegrationListDirectory extends AsyncComponent<
       return integration.projectList.length > 0 ? 2 : 0;
     }
 
-    if (!isSentryApp(integration)) {
-      return integrations.find(i => i.provider.key === integration.key) ? 2 : 0;
+    if (isSentryApp(integration)) {
+      const install = this.getAppInstall(integration);
+      if (install) {
+        return install.status === 'pending' ? 1 : 2;
+      }
+      return 0;
     }
 
-    const install = this.getAppInstall(integration);
-
-    if (install) {
-      return install.status === 'pending' ? 1 : 2;
+    if (isDocumentIntegration(integration)) {
+      return 0;
     }
 
-    return 0;
+    return integrations.find(i => i.provider.key === integration.key) ? 2 : 0;
   }
 
   getPopularityWeight = (integration: AppOrProviderOrPlugin) =>
     POPULARITY_WEIGHT[integration.slug] ?? 1;
 
   sortByName = (a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) =>
-    a.name.localeCompare(b.name);
+    a.slug.localeCompare(b.slug);
 
   sortByPopularity = (a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) => {
     const weightA = this.getPopularityWeight(a);
     const weightB = this.getPopularityWeight(b);
     return weightB - weightA;
   };
 
-  sortByInstalled = (a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) =>
-    this.getInstallValue(b) - this.getInstallValue(a);
+  sortByInstalled = (a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) => {
+    const out = this.getInstallValue(b) - this.getInstallValue(a);
+    return out;
+  };
 
   sortIntegrations(integrations: AppOrProviderOrPlugin[]) {
-    return integrations
-      .sort(this.sortByName)
-      .sort(this.sortByPopularity)
-      .sort(this.sortByInstalled);
+    return integrations.sort((a: AppOrProviderOrPlugin, b: AppOrProviderOrPlugin) => {
```

Comment:
The old way we were sorting was incorrect and was actually not working in tests but worked just fine in the app